### DISCUSSION TITLE
How to extract current timestamp in workflow?

### DISCUSSION THREAD

**Discussion by AmitGupta-appDirect** [Netflix/conductor#2897] [Q&A]
Hi,

In one of the requirements, we need to pass current timestamp in the body of an API call within HTTP type task.
Can you please suggest if there is a way to retrieve current timestamp in workflow.

We tried to add a JSON_JQ_TRANSFORM type task in the workflow to extract current timestamp whose result we could use in another HTTP type task to prepare http_request body, but it didn't work.

Is it even possible to extract/retrieve current timestamp in any of the task in workflow?

```
{
	"type": "JSON_JQ_TRANSFORM",
	"name": "json_transform",
	"taskReferenceName": "add_current_timestamp",
	"inputParameters": {
		"inputJSON": "${anotherHttpTask.output.response.body.customAttribute}",
		"queryExpression": ".inputJSON += [{ name: \\"current_timestamp\\", value: (now|strflocaltime("%d %m %Y %r %z"))}] "
	}
}
```

Thanks

**Comment by v1r3n** [comment]
Try the `INLINE` task where you can get the current time. Here is an example that gets the current date:

```
 {
      "name": "get_time",
      "taskReferenceName": "get_time",
      "inputParameters": {
        "evaluatorType": "javascript",
        "expression": "function e() { return {\"now\":  new Date().toJSON().slice(0,10).replace(/-/g,'/')}} e();"
      },
      "type": "INLINE",
      "decisionCases": {},
      "defaultCase": [],
      "forkTasks": [],
      "startDelay": 0,
      "joinOn": [],
      "optional": false,
      "defaultExclusiveJoinTask": [],
      "asyncComplete": false,
      "loopOver": []
    }
```
